tempCommit/PrintReads.wdl

```diff
@@ -0,0 +1,297 @@
+    interval_set_rule                                  = interval_set_rule,
+    lenient                                            = lenient,
+    QUIET                                              = QUIET,
+    read_filter                                        = read_filter,
+    read_index                                         = read_index,
+    read_validation_stringency                         = read_validation_stringency,
+    seconds_between_progress_updates                   = seconds_between_progress_updates,
+    sequence_dictionary                                = sequence_dictionary,
+    tmp_dir                                            = tmp_dir,
+    use_jdk_deflater                                   = use_jdk_deflater,
+    use_jdk_inflater                                   = use_jdk_inflater,
+    verbosity                                          = verbosity,
+
+  }
+
+  output {
+    # Workflow Outputs                                  
+    File PrintReadsoutput_arg = PrintReadsTask.PrintReadsTask_output_arg
+    File PrintReadsoutputcompanionIndex = PrintReadsTask.PrintReadsTask_outputcompanionIndex
+  }
+}
+
+task PrintReadsTask {
+
+  input {
+    String dockerImage
+    String gatk
+    String memoryRequirements
+    String diskRequirements
+    Array[String] input_arg
+    File output_arg
+    File outputcompanionIndex
+    Array[File]? arguments_file
+    Int? cloud_index_prefetch_buffer
+    Int? cloud_prefetch_buffer
+    Boolean? disable_bam_index_caching
+    Boolean? disable_sequence_dictionary_validation
+    Int? gcs_max_retries
+    String? gcs_project_for_requester_pays
+    Boolean? help
+    String? interval_merging_rule
+    Array[String]? intervals
+    File? reference
+    File? referenceDictionary
+    File? referenceIndex
+    Boolean? sites_only_vcf_output
+    Boolean? version_arg
+    Boolean? add_output_sam_program_record
+    Boolean? add_output_vcf_command_line
+    Boolean? create_output_bam_index
+    Boolean? create_output_bam_md5
+    Boolean? create_output_variant_index
+    Boolean? create_output_variant_md5
+    Array[String]? disable_read_filter
+    Boolean? disable_tool_default_read_filters
+    Array[String]? exclude_intervals
+    String? gatk_config_file
+    Int? interval_exclusion_padding
+    Int? interval_padding
+    String? interval_set_rule
+    Boolean? lenient
+    Boolean? QUIET
+    Array[String]? read_filter
+    Array[String]? read_index
+    String? read_validation_stringency
+    Float? seconds_between_progress_updates
+    String? sequence_dictionary
+    File? tmp_dir
+    Boolean? use_jdk_deflater
+    Boolean? use_jdk_inflater
+    String? verbosity
+
+  }
+
+  command <<<
+    ~{gatk} PrintReads \
+    --input_arg ~{sep=' --input_arg ' input_arg} \
+    --output_arg ~{sep=' --output_arg ' output_arg} \
+    ~{true='--arguments_file ' false='' defined(arguments_file)}~{sep=' --arguments_file ' arguments_file} \
```

Comment:
This is the best way to invoke arguments with multiple values, but I don't think this will work for any WDL variables that aren't of type `Array`: https://github.com/openwdl/wdl/blob/master/versions/1.0/SPEC.md#sep

Reply:
I've used something similar in the past for non-array args:

```
Input {
    String? max_read_length 
}
…
String max_read_length_arg = if defined(max_read_length) then " --max_read_length " else ""
...
command {
...
~{max_read_length_arg}~{default="" sep=" --max_read_length " max_read_length} 
...
}
```

I just didn't know there was something nicer with newer WDL versions.